astroid/nodes/as_string.py

```diff
@@ -624,11 +620,11 @@ def visit_matchstar(self, node: "MatchStar") -> str:
 
     def visit_matchas(self, node: "MatchAs") -> str:
         """Return an astroid.MatchAs node as string."""
-        # pylint: disable=import-outside-toplevel
-        # Prevent circular dependency
-        from astroid.nodes.node_classes import MatchClass, MatchMapping, MatchSequence
-
-        if isinstance(node.parent, (MatchSequence, MatchMapping, MatchClass)):
+        if node.parent.__class__.__name__ in (
+            "MatchSequence",
+            "MatchMapping",
+            "MatchClass",
+        ):
```

Comment:
I don't think this should be change. Although the current code isn't ideal either, it's still better than comparing the class name with predefined strings.

Reply:
We have to find something else then, it means redesigning the code in depth, right ?

Reply:
The proper way would probably be to resolve the circular dependency. To do that `NodeNG` can't import `as_string`. Thus `NodeNG.as_string()` should be set later.

Reply:
When you say that it should be set later do you mean using a postinit like in other node classes ? I fear this is the reason that importing astroid takes a long time and that pylint startup is slow now matter what.

Reply:
What I was thinking about was something similar to how we deal with the inference methods at the moment. I.e.
```py
# astroid/nodes/__init__.py

from astroid.nodes.node_ng import NodeNG
...
from astroid.as_string import AsStringVisitor

def as_string(self):
    return AsStringVisitor("    ")(self)

NodeNG.as_string = as_string
```

Tbh I like the current solution a bit better but that would be an alternative.

Reply:
I removed that so the only thing to merge now are the refactor that were done around it.